services/src/main/java/org/apache/druid/cli/CliPeon.java

```diff
@@ -355,6 +356,17 @@ public BroadcastDatasourceLoadingSpec getBroadcastDatasourcesToLoad(final Task t
           {
             return task.getBroadcastDatasourceLoadingSpec();
           }
+
+          @Provides
+          @LazySingleton
+          public LocalTmpStorage getLocalTmpStorage()
+          {
+            File tmpDir = new File(taskDirPath, "tmp");
```

Comment:
@cheddar : should we create the temporary storage per attempt? or should we share it across attempts for re-use of temp data?

Reply:
Let's share it.  The old default for cases where this would've been used is just `java.io.tmpdir` which was shared as well, so it shouldn't cause problems...